// antd-mobile/es/toast/demo/basic.js
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _createClass from 'babel-runtime/helpers/createClass';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
/* tslint:disable:no-console */
import React from 'react';
import { Toast, WhiteSpace, WingBlank, Button } from 'antd-mobile';
import { DeviceEventEmitter } from 'react-native';
function showToast() {
    Toast.info('这是一个 toast 提示!!!');
}
function successToast() {
    Toast.success('加载成功!!!', 1);
}
function showToastNoMask() {
    Toast.info('无 mask 的 toast !!!', 1, null, false);
}
function failToast() {
    Toast.fail('加载失败!!!');
}
function offline() {
    Toast.offline('网络连接失败!!!');
}
function loadingToast() {
    Toast.loading('加载中...', 1, function () {
        console.log('加载完成!!!');
    });
}

var ToastExample = function (_React$Component) {
    _inherits(ToastExample, _React$Component);

    function ToastExample() {
        _classCallCheck(this, ToastExample);

        var _this = _possibleConstructorReturn(this, (ToastExample.__proto__ || Object.getPrototypeOf(ToastExample)).apply(this, arguments));

        _this.alwaysShowToast = function () {
            Toast.info('这是一个 duration 为 0 的 toast!!!', 0);
            _this.timer = setTimeout(function () {
                Toast.hide();
            }, 5000);
        };
        return _this;
    }

    _createClass(ToastExample, [{
        key: 'componentDidMount',
        value: function componentDidMount() {
            DeviceEventEmitter.addListener('navigatorBack', function () {
                Toast.hide();
            });
        }
    }, {
        key: 'componentWillUnmount',
        value: function componentWillUnmount() {
            DeviceEventEmitter.removeAllListeners('navigatorBack');
            if (this.timer) {
                clearTimeout(this.timer);
                this.timer = null;
            }
        }
    }, {
        key: 'render',
        value: function render() {
            return React.createElement(
                WingBlank,
                { style: { marginTop: 80 } },
                React.createElement(WhiteSpace, null),
                React.createElement(
                    Button,
                    { onClick: showToastNoMask },
                    '\u65E0 mask'
                ),
                React.createElement(WhiteSpace, null),
                React.createElement(
                    Button,
                    { onClick: showToast },
                    '\u7EAF\u6587\u5B57 toast'
                ),
                React.createElement(WhiteSpace, null),
                React.createElement(
                    Button,
                    { onClick: successToast },
                    '\u6210\u529F toast'
                ),
                React.createElement(WhiteSpace, null),
                React.createElement(
                    Button,
                    { onClick: failToast },
                    '\u5931\u8D25 toast'
                ),
                React.createElement(WhiteSpace, null),
                React.createElement(
                    Button,
                    { onClick: offline },
                    '\u7F51\u7EDC toast'
                ),
                React.createElement(WhiteSpace, null),
                React.createElement(
                    Button,
                    { onClick: loadingToast },
                    '\u52A0\u8F7D\u4E2D toast'
                ),
                React.createElement(WhiteSpace, null),
                React.createElement(
                    Button,
                    { onClick: this.alwaysShowToast },
                    'duration \u4E3A 0 \u7684 toast'
                ),
                React.createElement(WhiteSpace, null)
            );
        }
    }]);

    return ToastExample;
}(React.Component);

export default ToastExample;
